autopilot/contractor.go

```diff
@@ -546,8 +546,30 @@ func (c *contractor) candidateHosts(cfg Config, wanted int) ([]consensus.PublicK
 		wanted = len(hosts)
 	}
 
-	// TODO: score each host and return selected hosts
-	return nil, nil
+	// score each host
+	scores := make([]float64, len(hosts))
+	for i, h := range hosts {
+		scores[i] = newHostScore(host{h, rhpv2.Contract{}}).
+			withAgeScore().
+			withCollateralScore(cfg).
+			withInteractionScore().
+			withSettingsScore(cfg).
+			withUptimeScore().
+			withVersionScore().
+			finalize()
```

Comment:
I don't think a fluent API is appropriate for host scoring. It's just a bunch of floats that we multiply together, right? Moreover, a fluent API implies that we would want to mix-and-match the scores -- but will we? IMO host scoring is not a very dynamic or flexible thing; we always use the same algorithm for every host.

Reply:
I decided to make it fluent to mirror the host filter and because I think we'll want a different scoring function when scoring potential new hosts vs existing hosts. The following scores don't make sense for potential new hosts:
- remain storage score
- age
- uptime

I know you could argue that it doesn't do any harm in including them in the score anyway because all hosts will have the same weight, but that's not really true for `age` and `uptime`. We might have had a host with which we had a contract for a day, and then for whatever reason dropped it, well it would get favoured only because it has an age. Pretty similar thing with uptime because it's not getting decayed over time.

I can change it though, just figured it's kind of nice if things behave similarly